test/plugins/javascript.test.js

```diff
@@ -29,39 +29,51 @@ describe('javascript-universal', () => {
   });
 
   it("resolves './modules/es6.symbol' without stripping .symbol suffix", () => {
-    assert.deepEqual(plugin.resolve({ target: './modules/es6.symbol' }), [
-      '{BASE_URL}modules/es6.symbol.js',
-      '{BASE_URL}modules/es6.symbol/index.js',
-      '{BASE_URL}modules/es6.symbol.jsx',
-      '{BASE_URL}modules/es6.symbol/index.jsx',
-      '{BASE_URL}modules/es6.symbol.ts',
-      '{BASE_URL}modules/es6.symbol/index.ts',
-      '{BASE_URL}modules/es6.symbol.tsx',
-      '{BASE_URL}modules/es6.symbol/index.tsx',
-      '{BASE_URL}modules/es6.symbol.ls',
-      '{BASE_URL}modules/es6.symbol/index.ls',
-      '{BASE_URL}modules/es6.symbol.json',
-      '{BASE_URL}modules/es6.symbol/index.json',
-      '{BASE_URL}modules/es6.symbol',
-    ]);
+    assert.deepEqual(
+      plugin.resolve({
+        target: './modules/es6.symbol',
+        path: '/user/repo/file.js',
```

Comment:
> EDIT: I looked into the test failures a little further, and I think we should be using path.posix.join instead of path.join, now that the tests are running outside of a browser context. See here for details: nodejs.org/api/path.html#path_windows_vs_posix

AppVeyor  is around for a while so it must be related to my changes. I had to pass `path` because I got the following error 

```
    TypeError: Path must be a string. Received undefined

      at javascriptFile (lib/plugins/javascript.js:24:52)
      at Object.resolve (lib/plugins/javascript.js:73:14)
      at Object.<anonymous> (test/plugins/javascript.test.js:33:26)
```

I wonder why this wasn't an issue before with Karma?

Reply:
I think we should switch from [path.resolve](https://nodejs.org/api/path.html#path_path_resolve_paths) to  [url.resolve](https://nodejs.org/api/url.html#url_url_resolve_from_to). Wonder why I didn't used `url.resolve` in the first place. I'll try that and let's see if the tests are still passing.

Reply:
Oh, `url.resolve` makes much more sense!

I'm also a little confused why `TypeError: Path must be a string. Received undefined` would just start happening now, but I guess it doesn't really matter.

Reply:
Mhhh, `url.resolve` doesn't work in that case. `url.resolve` resolves the given arguments similar to how a browser would resolve hrefs. This is problem, because we want to work with fs paths here.

The first row shows the current behaviour, where we use a combination of `path.dirname` and `path.join` to concatenate the path. The others are variations with `url.resolve`.

|input|output|
---|---
path.join(path.dirname('/some/folder/file.js'), '/views') | /some/folder/views
url.resolve('https://github.com/some/folder/file.js', '/views') | https://github.com/views
url.resolve('https://github.com/some/folder', '/views') | https://github.com/views
url.resolve('https://github.com/some/folder/', '/views') | https://github.com/views
url.resolve('https://github.com/some/folder/file.js', './views') | https://github.com/some/folder/views
url.resolve('https://github.com/some/folder/file.js', 'views') | https://github.com/some/folder/views

As you can see here, as long as the second argument `views` starts with a leading slash the url constructed url doesn't work for us. It feels like that this was the reason why I original used `path.join` over `url.resolve`.

Reply:
I pushed a fixup commit which reverts the changes I made to the tests. I'm passing an empty string for the path instead of a real path to avoid AppVeyor to fail. Not sure if this is what we want. At least it doesn't change the implementation

Reply:
Still failing on AppVeyor with many path issues 😞 I need to go to sleep, but will continue this journey tomorrow.

Reply:
As far as I understood, jest is using the real nodejs `path` module which takes the current environment into account when building paths. With karma this wasn't an issue, because webpack was including https://github.com/substack/path-browserify which doesn't take the current environment into account when building the path. Therefore it was working fine on AppVeyor. To me it feels stage to use `path.posix.join` just for the sake of fixing the tests in a windows environment. On the other hand, if we ignore this it would lead to failing tests for our windows contributors. I think we should `path.posix.join` a try.

Reply:
As expected `path.posix.join` fixed the windows build. @josephfrazier will merge the PR now